### DISCUSSION TITLE
Convenient way to change tabulation style(`hard_tabs` on/off).

### DISCUSSION THREAD

**Discussion by Motorenger** [zed-industries/zed#10895] [Feature Requests]
Right now I'm working on Python(either tab or space tabulation) and C(space tabulation) projects. And every time I switch between projects that have different tabulation style I need to go into the `settings.json` and enable/disable the `hard_tabs` option to use tab/spaces for tabulation, which is very inconvenient. If there no solution yet, I  propose to add another setting, which when enabled will allow Zed to decide the tabulation by itself depending on the file style. Even though it won't work for new/empty files, but at least once you started work, Zed then will catch up.

**Comment by talpadk** [comment]
Very new to zed, but maybe use per project settings?
(I know this isn't exactly what you are asking for)

From https://zed.dev/docs/configuring-zed :
"This configuration is merged with any local configuration inside your projects. You can open the project settings by running zed: open project settings from the command palette. This will create a .zed directory containing.zed/settings.json."

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
